### PR TITLE
nested: add support to telnet to serial port in nested VM

### PR DESCRIPTION
**Pull request by mvo5**
This commit adds support to use telnet to access the nested VM
serial port. This is useful for interactive debugging of the
nested VM if e.g. the creation of a user fails. It is also useful
to test/play-with console-conf in the nested env.

With that inside the L1 qemu one can type:
```
$ telnet localhost 7777
```
to get access to the serial port.

With "ctrl-]" and "send escape\nq\n" the telnet session can be
stopped again.

As a drive-by it cleans a duplicated PARAM_SERIAL definition. We
should address the duplication of all the PARAM_* for qemu in a
followup.

### CODE REVIEW

**Review by zyga — APPROVED** [review]
@mvo5 please remind me to annoy you with other hard problems, you are very good at it :-)

**Review by bboozzoo — APPROVED** [review]
LGTM. Do you think it's worth to add a monitor on another telnet port?

File: tests/lib/nested.sh [hunk 1]
```diff
@@ -593,7 +593,13 @@ nested_start_core_vm_unit() {
...
     PARAM_TRACE="-d cpu_reset"
     PARAM_LOG="-D $NESTED_LOGS_DIR/qemu.log"
-    PARAM_SERIAL="-serial file:${NESTED_LOGS_DIR}/serial.log"
+    # Open port 7777 on the host so that failures in the nested VM (e.g. to
+    # create users) can be debugged interactively via
+    # "telnet localhost 7777". Also keeps the logs
+    #
+    # XXX: should serial just be logged to stdout so that we just need
+    #      to "journalctl -u nested-vm" to see what is going on ?
+    PARAM_SERIAL="-chardev socket,telnet,host=localhost,server,port=7777,nowait,id=char0,logfile=${NESTED_LOGS_DIR}/serial.log -serial chardev:char0"
```
Comment (anonymouse64) [inline tests/lib/nested.sh:602]: this is great, I for some reason missed the `logfile` setting here when I tried to set this up with various pipes and running `tee` in the background, etc.

**Review by anonymouse64 — APPROVED** [review]
Thanks for this, I tried to make this work through various kinds of hacks with pipes and such without realizing there is a logfile setting, so great that you read the full manual and found that setting!

**Comment by mvo5** [comment]
Meh, looks like this needs some more work, the logfile optoin for qemu is not available on 16.04 it seems.

**Comment by anonymouse64** [comment]
mmm also if we continue to use logfile option in QEMU, maybe just omitting that on 16.04 or something, we should also set `logappend` so we get a logfile for every boot, but then we also need to delete the logfile when cleaning up after each run, not sure if we do that or not

File: tests/lib/nested.sh [hunk 2]
```diff
@@ -617,7 +617,18 @@ nested_start_core_vm_unit() {
     PARAM_CPU=""
     PARAM_TRACE="-d cpu_reset"
     PARAM_LOG="-D $NESTED_LOGS_DIR/qemu.log"
-    PARAM_SERIAL="-serial file:${NESTED_LOGS_DIR}/serial.log"
+    # Open port 7777 on the host so that failures in the nested VM (e.g. to
+    # create users) can be debugged interactively via
+    # "telnet localhost 7777". Also keeps the logs
+    #
+    # XXX: should serial just be logged to stdout so that we just need
+    #      to "journalctl -u nested-vm" to see what is going on ?
```
Comment (anonymouse64) [inline tests/lib/nested.sh:625]: then we need to handle the fact that for whatever reason when I tried this journald's default config loses some of the output when we turn on more verbose logging, so journald on the host VM needs to be configured differently I think to not lose earlier logs

perhaps the solution is simpler but I couldn't figure out why `journalctl -ef -u nested-vm` did not show the full log of the boot when I changed it to just `-serial mon:stdio`

**Review by anonymouse64 — APPROVED** [review]
lgtm, but one comment on the question inside

**Pull request merged**